# HG changeset patch
# User alta88@fixall.com
# Date 1729017000 -7200
# Parent  3e275ff4ce55229af358c3b849701096388ddb65
NNN13 - Fix internal dnd after failed external dnd:

Drag an unsuitable external file (image, PDF, etc.) onto a folder/thread and release despite "no parking sign".
Next internal drag&drop (message to folder, folder to folder) will fail.

diff --git a/mail/base/content/about3Pane.js b/mail/base/content/about3Pane.js
--- a/mail/base/content/about3Pane.js
+++ b/mail/base/content/about3Pane.js
@@ -3240,9 +3240,8 @@ var folderPane = {
   _onDragOver(event) {
     const systemDropEffect = event.dataTransfer.dropEffect;
 
     event.dataTransfer.dropEffect = "none";
-    event.preventDefault();
 
     const row = event.target.closest("li");
     this._timedExpand(row);
     if (!row) {
@@ -3417,8 +3416,9 @@ var folderPane = {
     } else {
       return;
     }
 
+    event.preventDefault();
     this._clearDropTarget();
     row.classList.add("drop-target");
   },
 
@@ -5384,10 +5384,8 @@ var threadPane = {
     if (event.target.closest("thead")) {
       this._clearDropTarget();
       return; // Only allow dropping in the body.
     }
-    // Must prevent default. Otherwise dropEffect gets cleared.
-    event.preventDefault();
     event.dataTransfer.dropEffect = "none";
     const types = Array.from(event.dataTransfer.mozTypesAt(0));
     const targetFolder = gFolder;
     if (targetFolder && types.includes("application/x-moz-file")) {
@@ -5415,8 +5413,10 @@ var threadPane = {
         row.classList.add("drop-target");
         this._currentDropRow = row;
       }
     }
+    // Must prevent default. Otherwise dropEffect gets cleared.
+    event.preventDefault();
   },
 
   /**
    * Handle threadPane drop events.
